source/scpp/src/crypto/XDRHasher.h

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+// Copyright 2018 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include <xdrpp/endian.h>
+#include <xdrpp/marshal.h>
+
+namespace stellar
+{
+
+// CRTP base class to use in XDR-hashing xdrpp-compatible archiver objects. All
+// you need to provide is a `void Derived::hashBytes(unsigned char const*,
+// size_t)` member.
+template <typename Derived> struct XDRHasher
```

Comment:
Do we need this (and shorthash / siphash) ?

Reply:
I think the problematic part is the QuorumIntersectionChecker. It's technically part of Herder and not SCP, so it brings in a lot of dependencies. For example:

ShortHash uses XdrHasher & SipHash
HashOfHash uses ShortHash
SecretKey uses HashOfHash
QuorumIntersectionChecker uses SecretKey